### DISCUSSION TITLE
Get custom field in Export Template

### DISCUSSION THREAD

**Discussion by rodvand** [netbox-community/netbox#9287] [Q&A]
Hi,

I have a custom field assigned to the Site model named orgnization. It is using the new object feature from NetBox 3.2, using the Tenant model. 
In the prefix list I want to create an Export Template which gets the prefix's site and lists the organisation assigned to it.

How can I get the custom field from a Site in this case? 

Let's say my Export template looks like this:
```
{% for prefix in queryset %}
{{ prefix.site.cf['organization'] }}
{% endfor %}
```

This works, but outputs the tenant ID in the list. Can I make a lookup somehow to get the name? In the REST API the full object is there so I suspect it's possible.

**Comment by jhernandfl** [comment]
Hi.  We also have a custom field defined as a Netbox object.  We would like the export template to display the name of the object, but so far, we can only figure out how to make the export template display the object's ID.  Any ideas?

**Comment by LuPo** [comment]
There is a method named `get_custom_fields()` available on objects:
```
    def get_custom_fields(self):
        """
        Return a dictionary of custom fields for a single object in the form {<field>: value}.
        """
```
First you need to get tenant ID as you already do. You will use it as an argument to deserialize() function to retrive your Tenant object.
 
Python version of this:
```
tenant_id = prefix.site.cf.get('organizzation')

cfields_site = prefix.site.get_custom_fields()
for cfield in cfields_site:
  if cfield.name == 'organizzation':
    site_org = cfield.deserialize(tenant_id)

print(site_org.name)
```
This is how you can do it in your Export Template:
```
{% for prefix in queryset %}
{%   for cfield in prefix.site.get_custom_fields() if cfield.name == 'organizzation' %}
{%     set site_org = cfield.deserialize(prefix.site.cf.get('organizzation')) %}
{{ site_org.name }}
{%   endfor %}
{% endfor %}
```

Hope it helps. If there is a better way, please let me know.
